Automattic/wp-calypso
# Checkout: Move eps payment method to wpcom-checkout

#### Changes proposed in this Pull Request

This PR moves the eps payment method from the `@automattic/composite-checkout` package into the `@automattic/wpcom-checkout` package. This is similar to how Bancontact was moved in https://github.com/Automattic/wp-calypso/pull/51690, Giropay in https://github.com/Automattic/wp-calypso/pull/53346, and p24 
in https://github.com/Automattic/wp-calypso/pull/53510 This is related to the refactoring of https://github.com/Automattic/wp-calypso/issues/52215

Note that the Eps payment method has been disabled in production since 2018. If we like, we could just remove it instead; see https://github.com/Automattic/wp-calypso/pull/53527 for that option.

<img width="580" alt="Screen Shot 2021-06-08 at 7 10 47 PM" src="https://user-images.githubusercontent.com/2036909/121269289-91621400-c88d-11eb-81ff-3db36c5a2848.png">


#### Background

The `composite-checkout` package is intended to be a fairly generic checkout-building toolkit. As part of this we originally envisioned it containing various payment methods, but that quickly became a problem as it became clear that the payment methods we use on WordPress.com are pretty specific to our platform. To reduce leaky abstractions, we moved many of our payment methods over to calypso, but some have remained because the cost of moving them seemed too high for the benefit.

As we prepare to publish version 1.0.0 of the `@automattic/composite-checkout` package on npm, I think it's a good time to clean up extraneous parts of the package. The payment methods are the only part of the package that are not fully TypeScript and are also still pretty specific to our WordPress.com infrastructure. Eventually it would be nice to convert them all to TypeScript and put them in a package like `@automattic/wpcom-checkout` (which is intended to be WordPress.com specific). This PR converts one such payment method and moves it there.

#### Testing instructions

- Apply D62510-code to your sandbox to force eps to be available.
- Set your user currency to EUR.
- Make a new purchase in checkout and select eps as the payment method, entering any string in the additional fields.
- Verify that you are redirected to the Stripe test page, that it reads "eps", and that the total, email, and name all match what you saw in checkout. (No need to complete the payment.)
